File: leetcode/Easy/0001-two-sum/0001-two-sum.cpp
#include <iostream>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> vals;
        for(int i=0;i<nums.size();i++){
            
            if(vals.find(nums[i]) != vals.end()){
                return {vals.at(nums[i]), i};
            }else{
                vals.insert({target-nums[i], i});
            }
        }
        return {0, 0};
    }
};

File: leetcode/Easy/0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        vals = {}
        for i in range(len(nums)):
            if nums[i] in vals:
                return [vals[nums[i]], i]
            else:
                vals[target - nums[i]] = i
